hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateNftsHandler.java

```diff
@@ -0,0 +1,138 @@
+import com.hedera.hapi.node.state.token.Nft;
+import com.hedera.hapi.node.token.TokenUpdateNftsTransactionBody;
+import com.hedera.hapi.node.transaction.TransactionBody;
+import com.hedera.node.app.service.token.ReadableTokenStore;
+import com.hedera.node.app.service.token.impl.WritableNftStore;
+import com.hedera.node.app.service.token.impl.validators.TokenUpdateNftValidator;
+import com.hedera.node.app.spi.fees.FeeContext;
+import com.hedera.node.app.spi.fees.Fees;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.spi.workflows.HandleException;
+import com.hedera.node.app.spi.workflows.PreCheckException;
+import com.hedera.node.app.spi.workflows.PreHandleContext;
+import com.hedera.node.app.spi.workflows.TransactionHandler;
+import com.hedera.node.config.data.TokensConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Provides the state transition for an NFT collection update.
+ */
+@Singleton
+public class TokenUpdateNftsHandler implements TransactionHandler {
+
+    private final TokenUpdateNftValidator tokenUpdateNftValidator;
+
+    @Inject
+    public TokenUpdateNftsHandler(@NonNull final TokenUpdateNftValidator tokenUpdateNftValidator) {
+        this.tokenUpdateNftValidator = tokenUpdateNftValidator;
+    }
+
+    @Override
+    public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
+        requireNonNull(txn);
+        final var op = txn.tokenUpdateNftsOrThrow();
+        validateTruePreCheck(op.hasToken(), INVALID_TOKEN_ID);
+    }
+
+    @Override
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+        final var op = context.body().tokenUpdateNftsOrThrow();
+        pureChecks(context.body());
+        final var tokenId = op.tokenOrThrow();
+        final var tokenStore = context.createStore(ReadableTokenStore.class);
+        final var tokenMetadata = tokenStore.getTokenMeta(tokenId);
+        if (tokenMetadata == null) throw new PreCheckException(INVALID_TOKEN_ID);
+    }
+
+    @Override
+    public void handle(@NonNull HandleContext context) throws HandleException {
+        requireNonNull(context);
+        final var txnBody = context.body();
+        final var op = txnBody.tokenUpdateNftsOrThrow();
+        final var tokenNftId = op.tokenOrThrow();
+
+        final var validationResult = tokenUpdateNftValidator.validateSemantics(context, op);
+        final var token = validationResult.token();
+        final var nftStore = context.writableStore(WritableNftStore.class);
+
+        // Wrap in Set to de-duplicate serial numbers
+        final var nftSerialNums = new ArrayList<>(new LinkedHashSet<>(op.serialNumbers()));
+        var nftCount = nftSerialNums.size();
+        validateTrue(nftCount <= nftStore.sizeOfState(), INVALID_TRANSACTION_BODY);
```

Comment:
You think `INVALID_NFT_ID` is appropriate here ? Since the `TransactionBody` is valid , but the serials provided are wrong